tensorflow/compiler/mlir/tensorflow/translate/import_model.cc

```diff
@@ -2599,8 +2599,31 @@ StatusOr<mlir::FunctionType> GraphDefImporter::InferMainFunctionType(
         return errors::InvalidArgument("Input ", i, "has invalid data type");
       }
     }
-    TF_RETURN_IF_ERROR(
-        ::tensorflow::ConvertDataType(imported_dtype, builder, &element_type));
+    /// Check if we have subtypes first
+    if (!node_info.subtypes.empty()) {
+      std::vector<mlir::TensorType> subtypes;
+      for (const auto& st : node_info.subtypes) {
+        mlir::Type t;
+        llvm::SmallVector<int64_t, 4> shape;
```

Comment:
Drop the explicit small vector shape unless you have a good reason to manually specify 4